ctapipe/image/showermodel.py

```diff
@@ -0,0 +1,102 @@
+import astropy.units as u
+import numpy as np
+from scipy.stats import multivariate_normal
+from scipy.spatial.transform import Rotation as R
+
+
+__all__ = [
+    "Gaussian",
+]
+
+
+class Gaussian:
+    @u.quantity_input(
+        x=u.m,
+        y=u.m,
+        phi=u.deg,
+        theta=u.deg,
+        first_interaction=u.m,
+        width=u.m,
+        length=u.m,
+    )
+    def __init__(self, n_ch, x, y, phi, theta, first_interaction, width, length):
+        """Create a 3D gaussian shower model for imaging.
+
+        Parameters
+        ----------
+        n_ch : int
+            Number of cherenkov photons in shower
+        x : u.Quantity[length]
+            x coord of shower intersection on ground
+        y : u.Quantity[length]
+            y coord of shower intersection on ground
+        phi : u.Quantity[angle]
+            azimuthal angle defining orientation of shower
+        theta : u.Quantity[angle]
+            polar angle defining orientation of shower
+        first_interaction : u.Quantity[length]
+            height of the first_interaction of the gamma above ground
+        width : u.Quantity[length]
+            width of the shower
+        length : u.Quantity[length]
+            length of the shower
+        """
+        self.n_ch = n_ch
+        self.x = x
+        self.y = y
+        self.phi = phi
+        self.theta = theta
+        self.first_interaction = first_interaction
+        self.width = width
+        self.length = length
+        self.barycenter = self.calcBC()
+
+    def density(self, x, y, z):
+        """Evaluates 3D gaussian with barycenter as the mean and width and height in the covariance matrix.
+        This matrix is rotated with the azimuthal and polar angle.
+        """
+        # Rotate covariance matrix
+        cov = np.zeros((3, 3)) * u.m
+        cov[0, 0] = self.width
+        cov[1, 1] = self.width
+        cov[2, 2] = self.length
+
+        r = R.from_rotvec([0, self.theta.to_value(u.rad), self.phi.to_value(u.rad)])
+        cov = r.as_matrix().T @ cov @ r.as_matrix()
+
+        gauss = multivariate_normal(
+            mean=self.barycenter.to_value(u.m), cov=cov.to_value(u.m)
+        )
+
+        return self.n_ch * gauss.pdf(np.array([x, y, z]))
+
+    def calcBC(self):
```

Comment:
code style: change to `calculate_barycenter()`  (avoid abbreviations, functions should be snake_case)

Reply:
or better yet, just make this a `functools.cached_property` called `barycenter`